cadc-uws-server/src/main/java/ca/nrc/cadc/uws/web/PostAction.java

```diff
@@ -0,0 +1,222 @@
+                        String edv = syncInput.getParameter(param);
+                        if (edv != null) {
+                            try {
+                                Long nv = new Long(edv);
+                                if (nv < job.getExecutionDuration()) {
+                                    job.setExecutionDuration(nv);
+                                    jobManager.update(jobID, job.getDestructionTime(), nv, job.getQuote());
+                                }
+                            } catch (Exception ex) {
+                                throw new IllegalArgumentException("invalid execution duration value (long): " + edv);
+                            }
+                        }
+                        break;
+                    case QUOTE:
+                        String qv = syncInput.getParameter(param);
+                        if (qv != null) {
+                            try {
+                                Date nv = df.parse(qv);
+                                if (nv.compareTo(job.getQuote()) < 0) {
+                                    job.setQuote(nv);
+                                    jobManager.update(jobID, job.getDestructionTime(), job.getExecutionDuration(), nv);
+                                }
+                            } catch (Exception ex) {
+                                throw new IllegalArgumentException("invalid destruction time value (IVOA timestamp): " + qv);
+                            }
+                        }
+                        break;
+                    default:
+                        throw new IllegalArgumentException("not modifiable: " + field);
+                }
+            }
+            
+            log.debug("redirect: " + redirectURL);
+            syncOutput.setHeader("Location", redirectURL);
+            syncOutput.setCode(303);
+        } catch (JobNotFoundException ex) {
+            throw new ResourceNotFoundException("not found: " + jobID, ex);
+        } catch (JobPersistenceException ex) {
+            throw new RuntimeException("failed to access job pertsistence", ex);
+        } finally {
+            log.debug("DONE: " + syncInput.getPath());
+        }
+    }
+    
+    private void doPhaseChange(String jobID) throws JobNotFoundException, JobPhaseException, JobPersistenceException, TransientException {
+        String nep = syncInput.getParameter("PHASE");
+        boolean run = "RUN".equalsIgnoreCase(nep);
+        boolean abort = "ABORT".equalsIgnoreCase(nep);
+        if (abort) {
+            jobManager.abort(jobID);
+        } else if (run) {
+            Job job = jobManager.get(jobID);
+            job.setProtocol(syncInput.getProtocol());
+            jobManager.execute(job);
+        } else {
+            throw new IllegalArgumentException("invalid PHASE value: " + nep);
+        }
+    }
+    
+    private boolean isDeleteAction() {
+        String action = syncInput.getParameter("ACTION");
```

Comment:
I didn't know UWS supported this action

Reply:
yeah, it was in the restlet code too - iirc it is a hack so a browser could perform job deletion